src/components/taskDetails/index.tsx

```diff
@@ -352,6 +352,10 @@ const TaskDetails: FC<Props> = ({ taskID }) => {
                                             data-testid="request-task-button"
                                             className={classNames.button}
                                             onClick={taskRequestHandle}
+                                            disabled={
+                                                taskDetailsData?.status.toUpperCase() ===
+                                                'ASSIGNED'
+                                            }
```

Comment:
instead of status go for if task details have an assignee or not because if status is in_progress then you can request for task

Reply:
Yes you can use UNASSIGNED or AVAILABLE status key for disabling the **Request Task** button.